sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1703,11 +1703,15 @@ passed an invalid event.
 The new handler methods, and queue shortcuts, defined by
 link:../supported/sycl_ext_oneapi_enqueue_barrier.asciidoc[sycl_ext_oneapi_enqueue_barrier]
 can only be used in graph nodes created using the Record & Replay API, as
-barriers rely on events to enforce dependencies. A synchronous exception will be
-thrown with error code `invalid` if a user tries to add them to a graph using
-the Explicit API. Empty nodes created with the `node::depends_on_all_leaves`
-property can be used instead of barriers when a user is building a graph with
-the explicit API.
+barriers rely on events to enforce dependencies. For barriers with an empty
+wait list parameter, the semantics are that the barrier node being added to
+will depend on all the existing graph leave nodes, not only the leave nodes
```

Comment:
I find barriers affecting nodes from other queues a bit counter-intuitive. I'm wondering what is the reasoning behind this change? Is it required to fix the bug mentioned in https://github.com/intel/llvm/issues/13066 or are we just clarifying that that's how the implementation works at the moment?

Reply:
Good question, this new wording is just clarifying the current implementation and is not related to the bugfix. However, when I was debugging the bugfix these semantics weren't immediately obvious so I thought best to state explicitly what's going on.

Reply:
Fair enough. I think that for this PR that sounds good. We could discuss the reasoning for the current implementation somewhere else.